### ISSUE TITLE
RTL with Editor , the editor RTL is not working 

### ISSUE THREAD

**Issue by wpanduka** [UXDivers/Grial-UI-Kit-Support#684]
I m having an issue with changing RTL with Editor , the editor RTL is not working , is their any help that i can get form your end ? screenshot attached . Is this a bug?? i will be waiting for positive reply , thank you in advance for your support 
![Screenshot_20200218-001554](https://user-images.githubusercontent.com/19517563/74679769-c3f18480-51e4-11ea-8f71-1da0e5fed7c3.jpg)

**Comment by dirivero** [comment]
Hi,

Can you try updating UXDivers.Grial package to version 3.0.61-RC? Make sure you check for pre-released nugets to see it.

After that you should only need to add this property to your Editor:

```
grial:EditorProperties.HorizontalTextAlignment="{ DynamicResource LayoutDirectionHorizontalTextAlignment }"
```

Let me know how it goes.

**Comment by wpanduka** [comment]
Thank you for your reply. I will test and let you know soon

**Comment by wpanduka** [comment]
Hi, Thank you we have checked with our solution by updating the version 3.0.61-RC and include ```
your code 
grial:EditorProperties.HorizontalTextAlignment="{ DynamicResource LayoutDirectionHorizontalTextAlignment }"
```
but we didn't get the expected result. the way we applied as shown below 

XAML code ( screenshot also attached ) 
![Editor](https://user-images.githubusercontent.com/19517563/75105365-104f2100-5638-11ea-95c6-a039129e30be.png)


```
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:grial="clr-namespace:UXDivers.Grial;assembly=UXDivers.Grial"
             NavigationPage.HasNavigationBar="False"
             NavigationPage.HasBackButton="False"
             FlowDirection="RightToLeft"
             x:Class="AABQatar.UserModule.Login.View.LoginPage"
             BackgroundColor="WhiteSmoke">
    <ContentPage.Content>
        <Grid>
            <Grid.RowDefinitions>
                <RowDefinition Height="*" />
                <RowDefinition Height="*" />

            </Grid.RowDefinitions>
            <Image  Grid.Row="0" Source="login_background1.png"
                    HorizontalOptions="EndAndExpand"
                    VerticalOptions="Start" Aspect="AspectFill"/>
            <StackLayout Grid.Row="1" Padding="28,5" VerticalOptions="Center"
                    BackgroundColor="Transparent">
                <Entry TextColor="Black" BackgroundColor="Transparent"
                        HeightRequest="52" FontSize="Medium"
                        FontFamily="{StaticResource AvenirMediumFont}"
                        Margin="0,0,0,5"
                        grial:EntryProperties.BorderColor="#d8d5d5"
                        grial:EntryProperties.BorderCornerRadius="6"
                        Placeholder="E-mail Address / Mobile Number"
                        Keyboard="Email"
                        Text="{Binding Username}"
                       />
                

                <Editor  grial:EditorProperties.Placeholder="TestABC"  grial:EditorProperties.PlaceholderColor="Black"
                    grial:EditorProperties.HorizontalTextAlignment="{ DynamicResource LayoutDirectionHorizontalTextAlignment }"
                         TextColor="Black" BackgroundColor="Transparent"
                        HeightRequest="52" FontSize="Medium"
                        FontFamily="{StaticResource AvenirMediumFont}" />
```

And the view model CS 
![csfile](https://user-images.githubusercontent.com/19517563/75105369-1e04a680-5638-11ea-997b-1b426ba0fc71.png)

```
public LoginViewModel(IApiResponseDeserializer _apiResponseDeserializer)
        {
            // ApiManager = _apiManager;
            ApiResponseDeserializer = _apiResponseDeserializer;
            LocalizationService.SetAppResourceCulture("Lan_Ar");
           bool _isArabic = LocalizationService.isArabic();
            //DialogService.ShowLoading("Loading",MaskType.Gradient);
            CommandLoginClick = new Command(async () => await LoginUser());
            ContinueAsGuestCommand = new Command(async () => await NavigateToGuestSignupOptPage());
            SignUpCommand = new Command(async () => await NavigateToSignupPage());
            ForgotPasswordCommand = new Command(async () => await NavigateToForgotPasswordPage());
        }
```
The out put screen as shown below 
![IMG-20200223-WA0003](https://user-images.githubusercontent.com/19517563/75105353-f7467000-5637-11ea-89e8-7157be200084.jpg)

As you can see above result we got and we expect the placeholder text to be placed in Right to left since we have already done in the xaml as well , so can you please help us what we should change or is their any thing missing in your code to achieve the goal we expect ?

**Comment by wpanduka** [comment]
Now in this way its working for Editor we have Crete a logic ( with flag) for the  `grial:EditorProperties.HorizontalTextAlignment="{ DynamicResource LayoutDirectionHorizontalTextAlignment }"` to make the TRL to move right when the culture changes  , will you be able to please create the same property for the Entry as well and Crete a release so we can download from the nuget and install it to use for the Entry fields. waiting for your favorable response

**Comment by dirivero** [comment]
In Grial, the value of `LayoutDirectionHorizontalTextAlignment` should change automatically depending on the culture (based on the layout direction of the culture: LTR or RTL).

The Entry already has that behavior but using the Xamarin.Forms native `HorizontalTextAlignment` property (editors don't have that property).

**Comment by IoTFier** [comment]
Hi @dirivero, how are you?

Unlike Grial's App, my App doesn't a switch for user to change LTR -> RTR and vice versa - rather user simply chooses language of preference in my app's settings for language preference, which 
in-turn will provide user the experience of LTR and RTR based on the chosen language.

In that case, **how to and where to set the value of LayoutDirectionHorizontalTextAlignment** in my App that uses Prism v7.2 MVVM when user sets / resets language preference in the App's preferences menu?

To be specific - **do I set the LayoutDirectionHorizontalTextAlignment (LTR/RTL) on the App loading in OnInitialized depending on user's language preference?** 

On first App install, the default language will be English, then user preference will be persisted locally or in the backend.

**how do I make sure the LTR <==> RTR is applied across App for ViewElements / Xaml coming from Grial kit or not ?**

Should I be setting RtlSwapHelper.Initialize as below ?
        
        protected override async void OnInitialized()
        {
             // I would like initialize CultureSettings using a bindable property which is set depending on the Mobile App's life cycle (default in the initial setup which is changed per user preference set up) 
             CultureSettings.Initialize("en", "ar");
            
             //what is the impact of that on the statement below ?
             //how do I make sure the LTR <==> RTR is applied across App for ViewElements / Xaml coming from Grial kit or not ?
             RtlSwapHelper.Initialize(UserLoginPage);

             await NavigationService.NavigateAsync("MenuPage/NavigationPage/LoginPage");
        }

Thank you

**Comment by dirivero** [comment]
Hi @IoTFier,

The values that depend on the layout direction are stored in specific resource dictionaries that are swapped at runtime.

The `LayoutDirectionHorizontalTextAlignment` is one of those resources. The runtime change is handled by the `RtlSwapHelper`.

**Comment by IoTFier** [comment]
Hi @dirivero , thank you for your input and helping me understand how Grial manages those resources. But, it doesn't address my specific questions around Prism MVVM implementation.  I am not using the master detail page implementation of Grial's App. 
 
Will you please help and guide me with on those questions above?  
I would really appreciate it. 

Thank you.

**Comment by dirivero** [comment]
> do I set the LayoutDirectionHorizontalTextAlignment (LTR/RTL) on the App loading in OnInitialized depending on user's language preference?

Yes, exactly.

> how do I make sure the LTR <==> RTR is applied across App for ViewElements / Xaml coming from Grial kit or not ?

Grial styles take care of the layout direction switch in pages, layouts (grid, stacks), native controls and Grial controls. If you have other custom controls you may need to add custom code/styles to achieve the switch. You have resources like `LayoutDirectionHorizontalTextAlignment` (among many others) to aid you, just remember to use them with `DynamicResource`.

**Comment by IoTFier** [comment]
Hi @dirivero, thank you for your help. But, I am still not clear on my understanding. 
Your guidance will be appreciated.

1. As I mentioned earlier, I am not using Grial's navigation but using Prism for MVVM.
    As a result, I am not using GetMainPage method to get the handle on the current page.
    If you know, then, after calling NavigationService.NavigateAsync("AppMenuPage/NavigationPage/LoginPage") - how and where do pass the handle of the current page to  RTLSwapHelper.Initialize  which is setting up LTR . RTR across the App ?

   How does this behavior works across page navigation using any MVVM framework such as Prism?
   Do I need call RTLSwapHelper.Initialize in the constructor of each page view model?

2. Do I have to update HorizontalTextAlignment for each XAML element across all ContentPages and ContentViews?  e.g. grial:EditorProperties.HorizontalTextAlignment="{ DynamicResource LayoutDirectionHorizontalTextAlignment }" 
    If so, that's a lot of code change. Can this be specified at the App level ?

Thank you.

**Comment by dirivero** [comment]
**Regarding 1**

`RTLSwapHelper` is only needed if you want to provide a navigation item that when pressed switches the layout direction. I understand that that is not your scenario so you can ignore that class completely.

`CultureSettings.Initialize("ar", "en");` is what you need to call. Make sure your app supports the cultures you pass in that method call.

**Regarding 2**

No, you don't. The global styles that come in the App.xaml take care of most of the scenarios. Are you using those Grial styles?

**Comment by IoTFier** [comment]
@dirivero ,
**Regarding 1**
As I send my code snippets to you in the email. I am providing options for users to select language preference in the app settings. That said, per Grial's set up I will nee to use RTLSwapHelper right ?
What I am not clear is in my app, users will have option to choose a language from a set of multiple languages offered in the App settings. 
In that case, how does the call CultureSettings.Initialize("en", "ar"); gets changed ?
I tried commenting if (IsCultureSupported(ci)) in "private static void UpdateLayoutDirectionDictionary(CultureInfo ci)" but then keep getting NPE after the command function on switch toggle is executed.

**Regarding 2**
I am using Grial Styles. But, on my custom nav bar the hamburger menu and right side icon are not getting swapped that are enclosed in a grid. I tried applying grial:Rtl.Mirror="True" & grial:Rtl.MirrorBehavior="Normal" on the Grid tag
